bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ItemChannelLinkRegistry.java

```diff
@@ -99,19 +198,40 @@ public void removeLinksForThing(ThingUID thingUID) {
 
     @Override
     protected void notifyListenersAboutAddedElement(ItemChannelLink element) {
-        super.notifyListenersAboutAddedElement(element);
-        postEvent(LinkEventFactory.createItemChannelLinkAddedEvent(element));
+        notifyListenersAboutAddedElement(element, itemRegistry.containsItem(element.getItemName()));
     }
 
     @Override
     protected void notifyListenersAboutRemovedElement(ItemChannelLink element) {
-        super.notifyListenersAboutRemovedElement(element);
-        postEvent(LinkEventFactory.createItemChannelLinkRemovedEvent(element));
+        // Workaround for removeProvider: see description of member 'active' above
+        final ItemRegistry itemRegistry = this.itemRegistry;
+        final boolean contains;
+        if (itemRegistry != null) {
+            contains = itemRegistry.containsItem(element.getItemName());
+        } else {
+            contains = true;
+        }
+        notifyListenersAboutRemovedElement(element, contains);
     }
 
     @Override
     protected void notifyListenersAboutUpdatedElement(ItemChannelLink oldElement, ItemChannelLink element) {
         super.notifyListenersAboutUpdatedElement(oldElement, element);
         // it is not needed to send an event, because links can not be updated
     }
+
+    protected void notifyListenersAboutAddedElement(final ItemChannelLink element, boolean itemExists) {
+        if (itemExists) {
+            super.notifyListenersAboutAddedElement(element);
```

Comment:
Okay, sound reasonable.

Should we inform the listeners on link creation and item creation (same question as above: should we use 'added'. 'removed' of the items here)?
Perhaps I misunderstand your comments I assumed you want to use that events in the channel link to refresh the channels, too.
Can we send the event twice on link creation and on item creation?